### DISCUSSION TITLE
How to normalize the edge features with GraphNorm?

### DISCUSSION THREAD

**Discussion by StefanIsSmart** [pyg-team/pytorch_geometric#8798] [Q&A]
The GraphNorm needs to put in the batch info of what you want to normalize.
We can use it to normalize the nodes' features between graphs easily.
But how to use it to norm the features of edges?

**Comment by rusty1s** [comment]
You can normalize edge features with it via
```python
norm(edge_feat, batch[edge_index[0]])

**Reply by abitofalchemy** [reply]
This works if I have multiple attributes on each edge? I want to normalize for each edge.

**Reply by rusty1s** [reply]
What do you mean? If you want to normalize for each edge individually, why not do
```
edge_attr = edge_attr.norm(p=2, dim=-1)
